### PR TITLE
Feature: lazy load checks on monitor dashboard

### PR DESCRIPTION
**Pull request by berezuev**
Feature: lazy loading of checks on frontend.

w/o lazy loading:
before_small
with lazy loading (gif is freezing. In real life it looks smoother):
after_small

### CODE REVIEW

File: Resources/public/javascript/app.js [hunk 1]
```diff
@@ -41,16 +41,29 @@ Health.healthController = Em.ArrayProxy.create({
     content: [],
 
     runChecks: function() {
-        var self = this;
+        let self = this;
+        let lastResponseLength = false;
         jQuery.ajax({
-            url: api.liip_monitor_run_all_checks,
+            url: 'http://localhost:8000/_health/stream',
```
Comment (berezuev) [inline Resources/public/javascript/app.js]: todo: replace with alias

**Comment by berezuev** [comment]
@kbond WDYT?

**Comment by kbond** [comment]
Looks good @berezuev - can you make it opt-in?

**Comment by berezuev** [comment]
@kbond, I've made it optional, but set parameter true by default. I think, it would work in all actual browsers. 

Not sure about documentation, English is not my native language. Can you check it, please?

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -55,6 +55,9 @@ public function getConfigTreeBuilder()
                     ->min(100)->max(598)
                     ->defaultValue(502)
                 ->end()
+                ->booleanNode('is_lazy_run')
```
Comment (kbond) [inline DependencyInjection/Configuration.php]: Can you add and `->info()` here to explain the feature?

File: README.md [hunk 3]
```diff
@@ -259,6 +259,7 @@ groups. Additionally, the `monitor:list` has an option `--groups` to list all re
 ```yml
 liip_monitor:
     enable_controller:    false
+    is_lazy_ryn:          true
```
Comment (kbond) [inline README.md]: The info added in `Configuration` should be added here as well.

File: Resources/views/health/index.html.php [hunk 4]
```diff
@@ -172,6 +172,12 @@
 }</pre>
         </dd>
 
+        <dt><a href="<?php echo $request->getUriForPath($request->getPathInfo().'stream'); ?>"><?php echo $request->getPathInfo().'stream'; ?></a></dt>
+        <dl>Performs all health checks, and lazily streams the results as an array of JSON objects. Works like <code><?php echo $request->getPathInfo().'run'; ?></code>,
+            but checks show right after it finish. Don't use it with standard json parsers. <br/>
+            If <code>is_lazy_run</code> parameter is <strong>true</strong> (by default), checks loads using this method. Overwise uses <code><?php echo $request->getPathInfo().'run'; ?></code>.
```
Comment (kbond) [inline Resources/views/health/index.html.php]:
```suggestion
            If <code>is_lazy_run</code> parameter is <strong>true</strong> (by default), checks load using this method. Overwise uses <code><?php echo $request->getPathInfo().'run'; ?></code>.
```

**Comment by kbond** [comment]
@lsmith77 what do you think of this? Specifically making it the default? I don't use the frontend provided in this bundle for any projects so I'm apprehensive about merging. Do you know anyone who uses it that could review?

File: README.md [hunk 5]
```diff
@@ -259,6 +259,7 @@ groups. Additionally, the `monitor:list` has an option `--groups` to list all re
 ```yml
 liip_monitor:
     enable_controller:    false
+    is_lazy_run:          true #Load checks right after it finished using /stream method
```
Comment (kbond) [inline README.md]:
```suggestion
    is_lazy_run:          true # Load checks right after it finished using /stream method
```

**Comment by lsmith77** [comment]
as long as its configurable its fine for me.

**Comment by kbond** [comment]
Thanks @berezuev!

**Comment by kbond** [comment]
Released in [2.11.0](https://github.com/liip/LiipMonitorBundle/releases/tag/2.11.0)

**Comment by berezuev** [comment]
@kbond I'm really sorry, but i think, you'd better revert this. I've been testing it using php dev server (php -S), and now I've tried in on nginx+php-fpm. Flushing echo doesn't work properly here (because of caching, gzipping, etc).

**Comment by kbond** [comment]
No problem, I reverted in #225

**Pull request merged**